internal/util/cmdutil/cmdutil.go

```diff
@@ -150,7 +150,17 @@ func WriteFnOutput(dest, content string, fromStdin bool, w io.Writer) error {
 func WriteToOutput(r io.Reader, w io.Writer, outDir string) error {
 	var outputs []kio.Writer
 	if outDir != "" {
-		err := os.MkdirAll(outDir, 0700)
+		_, err := os.Stat(outDir)
+
+		if err == nil || os.IsExist(err) {
+			return fmt.Errorf("directory %q already exists, please delete the directory and retry", outDir)
```

Comment:
add a e2e test that fails and assert this failure for eval, render, and sink. Failure cases are as important (or even more important) than happy paths.

Reply:
Added.